lms/djangoapps/course_api/serializers.py

```diff
@@ -0,0 +1,104 @@
+"""
+Course API Serializers.  Representing course catalog data
+"""
+
+import urllib
+
+from django.core.urlresolvers import reverse
+from django.template import defaultfilters
+
+from rest_framework import serializers
+
+from xmodule.course_module import DEFAULT_START_DATE
+from lms.djangoapps.courseware.courses import course_image_url, get_course_about_section
+
+
+class _MediaSerializer(serializers.Serializer):
+    def __init__(self, name=u'', description=u'', type=u'', uri_parser=lambda x: x, *args, **kwargs):
+        super(_MediaSerializer, self).__init__(*args, **kwargs)
+
+        self.name_value = name
+        self.description_value = description
+        self.type_value = type
+        self.uri_parser = uri_parser
+
+    uri = serializers.SerializerMethodField(source='*')
+    name = serializers.SerializerMethodField()
+    description = serializers.SerializerMethodField()
+    type = serializers.SerializerMethodField()
+
+    def get_name(self, _):
+        return self.name_value
+
+    def get_description(self, _):
+        return self.description_value
+
+    def get_type(self, _):
+        return self.type_value
+
+    def get_uri(self, course):
+        '''
+        Get the representation for SerializerMethodField `course_image`
+        '''
+        return self.uri_parser(course)
+
+
+class _MediaCollectionSerializer(serializers.Serializer):
+    image = _MediaSerializer(source='*', name=u'Course Image', description=u'', type=u'logo', uri_parser=course_image_url)
+
+
+class CourseSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for Course objects
+    """
+
+    course_id = serializers.CharField(source='id', read_only=True)  # pylint: disable=invalid-name
+    name = serializers.CharField(source='display_name_with_default')
+    number = serializers.CharField(source='display_number_with_default')
+    org = serializers.CharField(source='display_org_with_default')
+    short_description = serializers.SerializerMethodField()
+    media = _MediaCollectionSerializer(source='*')
+    start = serializers.DateTimeField()
+    start_type = serializers.SerializerMethodField()
+    start_display = serializers.SerializerMethodField()
+    end = serializers.DateTimeField()
+    enrollment_start = serializers.DateTimeField()
+    enrollment_end = serializers.DateTimeField()
+    blocks_url = serializers.SerializerMethodField()
```

Comment:
@cpennington I've reworked the serialization to match the ecommerce api naming scheme a bit better.  @nasthagiri wanted to revisit course_id vs id vs course_key vs. usage_key and we wanted to spend a little more time figuring out how to handle urls in a more generic and predictable fashion.  I was also wondering if the "type" field in the media representations was supposed to be analogous to the IANA standard rel fields, or share those semantics? See https://www.iana.org/assignments/link-relations/link-relations.xhtml for standard examples.